frontend/src/components/pages/InitialRegisterSetupPage/InitialRegisterSetupPage.tsx

```diff
@@ -0,0 +1,46 @@
+import { supabase } from "@/supabaseclient";
+import { useEffect } from "react";
+
+export default function InitialRegisterSetupPage() {
+  {
+    /*In here is where fetch requesto backend to setup profile is going to take plac. You will send in Authorization header the token we get from session above  */
+  }
+
+  useEffect(() => {
+    async function checkUserProfile() {
+      try {
+        const sessionResult = await supabase.auth.getSession();
```

Comment:
You don't need to request the session on every load, I let more details on the general comment